### PR TITLE
Fix build copy for object files

### PR DESCRIPTION
**Pull request by hmahmood**
### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by paulcacheux — APPROVED** [review]

**Review by brycekahle — APPROVED** [review]

**Comment by hmahmood** [comment]
@paulcacheux @brycekahle also reverted use of `ctx.sudo` since was not working (wouldn't take any input). See note here: https://docs.pyinvoke.org/en/stable/api/context.html#invoke.context.Context.sudo

**Comment by brycekahle** [comment]
I think you can get rid of the use of `ctx.sudo` in `clean_object_files` too, since `ninja` shouldn't be creating any files owned by root anymore.

File: tasks/system_probe.py [hunk 1]
```diff
@@ -833,10 +833,10 @@ def build_object_files(
     ctx.run(f"ninja -d explain -f {nf_path}")
 
     if not windows:
-        runner = ctx.run if is_root() else ctx.sudo
-        runner(f"mkdir -p {EMBEDDED_SHARE_DIR}")
-        runner(f"cp -R {build_dir} {EMBEDDED_SHARE_DIR}")
-        runner(f"chown root:root -R {EMBEDDED_SHARE_DIR}")
+        sudo = "" if is_root() else "sudo"
+        ctx.run(f"{sudo} mkdir -p {EMBEDDED_SHARE_DIR}")
+        ctx.run(f"{sudo} cp -R {build_dir} {EMBEDDED_SHARE_DIR}")
```
Comment (paulcacheux) [inline tasks/system_probe.py:838]: Don't you need the `*` ?

**Comment by paulcacheux** [comment]
Just found out I have a
```
bpf_dir: /home/vagrant/go/src/github.com/DataDog/datadog-agent/pkg/ebpf/bytecode/build
```
in my `system-probe.yaml` which explains why it works perfectly fine for me without the cp stuff (or with a wrong cp). Thanks for fixing it.

As a side not, it seems we all have kind of different dev env so this kind of stuff is always tricky to get right for everyone

File: tasks/system_probe.py (hunk 1 above)
Comment (hmahmood) [reply tasks/system_probe.py]: Yes, that was the original fix that was overwritten by me cherry-picking the sudo fix from another branch I have -- thanks for catching that.

**Pull request merged**